enterprise/views.py

```diff
@@ -952,3 +964,376 @@ def get(self, request, enterprise_uuid, course_id):
+            'welcome_text': self.context_data['welcome_text'].format(platform_name=platform_name),
+            'enterprise_welcome_text': self.context_data['enterprise_welcome_text'].format(
+                strong_start='<strong>',
+                strong_end='</strong>',
+                enterprise_customer_name=enterprise_customer.name,
+                platform_name=platform_name,
+            ),
+            'page_title': self.context_data['page_title'].format(item=item),
+            'program_title': program_details['title'],
+            'program_subtitle': program_details['subtitle'],
+            'program_overview': program_details['overview'],
+            'program_price': program_enrollment_details['program_price'],
+            'program_discounted_price': program_enrollment_details['program_discounted_price'],
+            'courses': program_enrollment_details['courses'],
+            'item_bullet_points': self.context_data['item_bullet_points'],
+            'purchase_text': self.context_data['purchase_text'].format(purchase_action=purchase_action),
+            'discount_provider': discount_provider,
+            'expected_learning_items': program_details['expected_learning_items'],
+            'course_count_text': course_count_text,
+            'price_info_text': self.context_data['price_info_text'].format(
+                purchase_text=self.context_data['purchase_text'].format(
+                    purchase_action=purchase_action.lower(),
+                )
+            ),
+            'length_info_text': length_info_text,
+            'effort_info_text': effort_info_text,
+            'is_program_eligible_for_one_click_purchase': program_details['is_program_eligible_for_one_click_purchase'],
+        })
+        return render(request, 'enterprise/enterprise_program_enrollment_page.html', context=context_data)
+
+    @method_decorator(force_fresh_session)
+    @method_decorator(enterprise_login_required)
+    def get(self, request, enterprise_uuid, program_uuid):
+        """
+        Show Program Landing page for the Enterprise's Program.
+
+        Render the Enterprise's Program Enrollment page for a specific program.
+        The Enterprise and Program are both selected by their respective UUIDs.
+
+        Unauthenticated learners will be redirected to enterprise-linked SSO.
+
+        A 404 will be raised if any of the following conditions are met:
+            * No enterprise customer UUID query parameter ``enterprise_uuid`` found in request.
+            * No enterprise customer found against the enterprise customer
+                uuid ``enterprise_uuid`` in the request kwargs.
+            * No Program can be found given ``program_uuid`` either at all or associated with
+                the Enterprise..
+        """
+        verify_edx_resources()
+        enterprise_customer, program_details, program_enrollment_details = self.get_base_details(
```

Comment:
I believe we need to do this _after_ we do the EnterpriseCustomerUser creation; otherwise we won't get accurate entitlement details for the user based on Enterprise linkage from the Ecommerce API. That's the whole reason behind making the view nonatomic in the first place.

Reply:
:+1: